File: app/src/main/java/com/teamunemployment/lolanalytics/FrontPage/Tabs/MatchHistoryTab/Dependencies/MatchHistoryModule.java
package com.teamunemployment.lolanalytics.FrontPage.Tabs.MatchHistoryTab.Dependencies;

import android.content.Context;

import com.teamunemployment.lolanalytics.io.RESTApiExecutor;
import com.teamunemployment.lolanalytics.io.RealmExecutor;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.MatchHistoryTab.MatchHistoryInteractor;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.MatchHistoryTab.MatchHistoryPresenter;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.StatsComparisonTab.Model.BarChartModel;

import dagger.Module;
import dagger.Provides;

/**
 * @author Josiah Kendall
 */

@Module

public class MatchHistoryModule {
    @Provides
    MatchHistoryPresenter provideMatchHistoryBasePresenter(MatchHistoryInteractor matchHistoryInteractor, BarChartModel barChartModel) {
        return new MatchHistoryPresenter(matchHistoryInteractor, barChartModel);
    }
    // Need context for static Realm initialisation
    @Provides
    MatchHistoryInteractor provideMatchHistoryModel(RESTApiExecutor restApiExecutor, RealmExecutor realmExecutor, Context context) {
        return new MatchHistoryInteractor(restApiExecutor, realmExecutor, context);
    }
}

File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/tabs/MatchHistoryTab/MatchHistoryModule.kt
package com.teamunemployment.lolanalytics.front_page.tabs.MatchHistoryTab

import com.teamunemployment.lolanalytics.front_page.tabs.MatchHistoryTab.Model.BarChartModel
import org.koin.android.module.AndroidModule

/**
 * @author Josiah Kendall
 */
class MatchHistoryModule : AndroidModule() {
    override fun context() = applicationContext {

        // TODO
        context(name = "MatchHistoryModule") {
            provide { MatchHistoryPresenter(get(), get(), get(), get()) }
            provide { MatchHistoryInteractor(get(), get(), get(), get(), get()) }
            provide { BarChartModel() }
            provide { MatchHistoryRecycleViewAdapter()}
        }
    }
}